http-server/auth/auth.go

```diff
@@ -0,0 +1,159 @@
+package authorization
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	http_server "github.com/ZhdanovichVlad/go_final_project/http-server"
+	"github.com/ZhdanovichVlad/go_final_project/storage/users"
+	"github.com/dgrijalva/jwt-go"
+	"net/http"
+	"os"
+	"time"
+)
+
+type Claims struct {
+	Authorized bool `json:"authorized"`
+	Sum        int  `json:"sum"`
+	jwt.StandardClaims
+}
+
+// Authorization function for user authorization.
+// Authorization функция для авторизации пользователя.
+func Authorization(w http.ResponseWriter, req *http.Request) {
+	var user users.User
+	var buf bytes.Buffer
+	_, err := buf.ReadFrom(req.Body)
+	if err != nil {
+		msg, errInt := http_server.JsonErrorMarshal(http_server.TaskResponseError{"Error when reading from req.Body"}, true)
```

Comment:
Что сразу бросается в глаза - последний параметр с `isBadRequest`, в реальных проектах может быть много типом ошибок, а тут мы просто через флаг передаем в IF какой нам нужен код. Я бы убрал эту штуку из функции и просто в следующей строчке явно писал какой код возвращаем.

Reply:
Так же на входе можно принимать просто `string`, а внутри уже конвертировать
```suggestion
		msg, errInt := http_server.JsonErrorMarshal("Error when reading from req.Body", true)
```